src/js/components/Calendar/Calendar.js

```diff
@@ -200,59 +199,67 @@ export const getOutputFormat = (dates) => {
 
 const millisecondsPerYear = 31557600000;
 
-const CalendarDay = ({
-  children,
-  fill,
-  size,
-  isInRange,
-  isSelected,
-  otherMonth,
-  rangePosition,
-  buttonProps = {},
-  responsive,
-}) => {
-  const { passThemeFlag } = useThemeValue();
-  const usingKeyboard = useKeyboard();
-
-  return (
-    <StyledDayContainer
-      role="gridcell"
-      inRange={isInRange}
-      isSelected={isSelected}
-      rangePosition={rangePosition}
-      sizeProp={size}
-      fillContainer={fill}
-      responsive={responsive}
-    >
-      <StyledDayButton
-        fill={fill}
-        tabIndex={-1}
-        plain
+const CalendarDay = forwardRef(
+  (
+    {
+      activeProp,
+      children,
+      day,
+      disabledProp,
+      fill,
+      size,
+      isInRange,
+      isSelected,
+      otherMonth,
+      rangePosition,
+      buttonProps = {},
+      responsive,
+    },
+    ref,
+  ) => {
+    const { passThemeFlag } = useThemeValue();
+    return (
+      <StyledDayContainer
+        inRange={isInRange}
+        isSelected={isSelected}
+        rangePosition={rangePosition}
+        sizeProp={size}
+        fillContainer={fill}
         responsive={responsive}
-        {...buttonProps}
       >
-        {({ active, hover }) => (
-          <StyledDay
-            // only apply active styling when using keyboard
-            // otherwise apply hover styling
-            active={usingKeyboard ? active : undefined}
-            disabledProp={buttonProps.disabled}
-            hover={hover}
-            inRange={isInRange}
-            isSelected={isSelected}
-            otherMonth={otherMonth}
-            sizeProp={size}
-            fillContainer={fill}
-            responsive={responsive}
-            {...passThemeFlag}
-          >
-            {children}
-          </StyledDay>
-        )}
-      </StyledDayButton>
-    </StyledDayContainer>
-  );
-};
+        <StyledDayButton
+          role="gridcell"
+          aria-current={activeProp?.getTime() === day?.getTime()}
+          aria-disabled={disabledProp}
+          aria-selected={isSelected}
+          disabledProp={disabledProp}
```

Comment:
```suggestion
          disabled={disabledProp}
```
Should this be passed to `disabled` so it's passing through as a button prop?

Reply:
I didn't want the button to get the disabled prop so that we could still navigate to it and instead use aria-disabled to signify that it can't be interacted with. See https://github.com/grommet/grommet/pull/7612#discussion_r2073752541

Reply:
I see -- let's continue this question in the comment thread you linked!